compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -30,6 +30,10 @@ pub struct ModuleItems {
 }
 
 impl ModuleItems {
+    /// Returns all non-associated locally defined items in all modules.
+    ///
+    /// Note that this does *not* include associated items of `impl` blocks! It also does not
+    /// include foreign items. If you want to e.g. get all functions, use `definitions()` below.
     pub fn items(&self) -> impl Iterator<Item = ItemId> + '_ {
```

Comment:
maybe rename it to `free_items`?

Reply:
Does that term have any precedent in the compiler?

Reply:
there are 56 results for `free function` and 6 results for `free const`, among them results in diagnostic messages.

Reply:
and `resolve_instance` refers to `free item` in debug logging

Reply:
Okay. :)

"free" captures "non-assoc" but doesn't capture "also excludes foreign items". But I think that's less important to capture anyway.

Reply:
Also it seems to include the impl blocks themselves, or did I misunderstand that?